### PR TITLE
adds credit to restapi and machinelearning

### PR DESCRIPTION
**Pull request by yati1998**

### CODE REVIEW

**Comment by yati1998** [comment]
@alodahl I have made the changes. Please do review it and mention if you want some other changes to be done.

**Comment by yati1998** [comment]
@alodahl I have made the changes. Please do review it and mention if you want some other changes to be done.

File: api/REST-API.md [hunk 1]
```diff
@@ -50,7 +50,8 @@ Stateless  client/server  protocol:  each  HTTP  contains  all  the  necessary
...
 - Layered  system – The  layered  system  style  allows  an  architecture  to  be  composed  of  hierarchical  layers  by  constraining  component behavior  such  that  each  component  cannot  “see”  beyond  the  immediate  layer  with  which  they  are  interacting.
-- Code  on  demand – REST  allows  client  functionality  to  be  extended  by  downloading  and  executing  code  in  the  form  of applets  or  scripts.  This  simplifies  clients  by  reducing  the  number  of  features  required  to  be  pre-implemented.
+- Code  on  demand – REST  allows  client  functionality  to  be  extended  by  downloading  and  executing  code  in  the  form  of applets  or  scripts.  This  simplifies  clients  by  reducing  the  number  of  features  required  to  be  pre-implemented.[(restfulapi.net)](https://restfulapi.net/)
```
Comment (alodahl) [inline api/REST-API.md]: This report said there were more sources... is this really all from that same site?  https://smallseotools.com/view-report/55935ef38455fd043acbc748820c891e/

File: roles-in-tech/machine-learning.md [hunk 2]
```diff
@@ -33,7 +33,7 @@ In this instance, the machine – or algorithm – was developed by DeepMind, wh
...
 ### 4.Microsoft
  - Microsoft has actually been the third-biggest spender on acquisitions over the past five years, according to CB Insights.
-The company is well and truly into the internet market, especially after its $26 billion purchase of LinkedIn a couple of years ago. LinkedIn may provide the best platform for Microsoft to showcase the enterprise applications it develops based on machine learning.
+The company is well and truly into the internet market, especially after its $26 billion purchase of LinkedIn a couple of years ago. LinkedIn may provide the best platform for Microsoft to showcase the enterprise applications it develops based on machine learning.[(em360tech.com)](https://www.em360tech.com/tech-news/top-ten/top-10-companies-using-machine-learning/)
```
Comment (alodahl) [inline roles-in-tech/machine-learning.md]: Thank you! Cite Bailey too please

File: api/REST-API.md (hunk 1 above)
Comment (yati1998) [reply api/REST-API.md]: @alodahl yes I am sure that this part is also from the same source. it's the last point in the list in the given link. I have removed the word optional from the description and hence it must be showing some other sites too. But I took this data from the site I have mentioned.

File: roles-in-tech/machine-learning.md (hunk 2 above)
Comment (yati1998) [reply roles-in-tech/machine-learning.md]: @alodahl  I am sorry but couldn't get what to do. Can you please explain what do you want me to do?

**Comment by yati1998** [comment]
@alodahl what change do you want me to do now?

File: roles-in-tech/machine-learning.md (hunk 2 above)
Comment (alodahl) [reply roles-in-tech/machine-learning.md]: Oops, that must have been autocorrected.  Cite Baidu please.

**Comment by yati1998** [comment]
@alodahl I have added the source to Baidu. Do you want any further changes or is it done?

File: roles-in-tech/machine-learning.md [hunk 3]
```diff
@@ -30,10 +30,11 @@ Whether it’s something it acquired or developed in-house, Google has launched
...
-The company is particularly interested in natural language processing, with a view to developing a workable voice-activated search function. One of the many machine learning acquisitions Baidu made last year was [Kitt.ai](http://kitt.ai/), which is said to have a portfolio of chatbots and voice-based applications. The financial terms of that particular deal were not disclosed but Baidu is said to be the 10th biggest-spender on acquisitions in the world, according to CB Insights, which says Google spends the most.
+The company is particularly interested in natural language processing, with a view to developing a workable voice-activated search function. One of the many machine learning acquisitions Baidu made last year was [Kitt.ai](http://kitt.ai/), which is said to have a portfolio of chatbots and voice-based applications. The financial terms of that particular deal were not disclosed but Baidu is said to be the 10th biggest-spender on acquisitions in the world, according to CB Insights, which says Google spends the most.[(em360tech.com)](https://www.em360tech.com/tech-news/top-ten/top-10-companies-using-machine-learning/)
```
Comment (alodahl) [inline roles-in-tech/machine-learning.md]: Do the prerequisites, motivation, or ibm sections have their source?

Comment (yati1998) [reply roles-in-tech/machine-learning.md]: @alodahl IBM  and Google have the same source as that of the others. Shall I mention again individually?
The motivation section was not written by me, it was already there. and prerequisites have the source.

Comment (alodahl) [reply roles-in-tech/machine-learning.md]: Yes, mention ibms individually... linked to its own part of the page if you can.

The prereq has a link, but does the paragraph that it is in come from that source?

Comment (yati1998) [reply roles-in-tech/machine-learning.md]: yes, the link mentioned in prerequisites is for its description. there are 2 links one for the definition of data analysis and other from where the paragraph is cited.

Comment (yati1998) [reply roles-in-tech/machine-learning.md]: and about individual links, its all in the same page and doesn't have specific individual links. So I am mentioning the same in all.

**Comment by yati1998** [comment]
@alodahl I have made the changes. I hope you will like it this time.

**Comment by yati1998** [comment]
@alodahl I have cited the source for the machine learning motivation section.

**Comment by alodahl** [comment]
Where is it btbp cited? 
image

**Comment by alodahl** [comment]
Thanks Yati!

**Pull request merged**